### DISCUSSION TITLE
parse_cef translate_custom_fields

### DISCUSSION THREAD

**Discussion by aacgood** [vectordotdev/vrl#818] [Q&A]
Hi,

Im using the `parse_cef` function with `translate_custom_fields: true` and it seems like its not remapping all my cs#Label / cs# fields.  

Having a look at the source code, it appears that `build_map` only has a hard coded list of 22 fields it will do the translation on.  (ie: it doesnt do cs7Label/cs7 onwards)

https://github.com/vectordotdev/vrl/blob/main/src/stdlib/parse_cef.rs#L15

Is this expected, or is there a way to have the code just look for cs?Label -> cs? for logs that contains a large number of custom fields and not be limited to whats in the current build_map?

**Comment by aacgood** [comment]
Anyone able to comment on this?

**Comment by pront** [comment]
Hi @aacgood, can you provide an example in the VRL playground? 
Something like [this](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLiA9IHBhcnNlX2NlZiEoLm1lc3NhZ2UpIiwiZXZlbnQiOnsibWVzc2FnZSI6IkNFRjowfEluY2Fwc3VsYXxTSUVNaW50ZWdyYXRpb258MXwxfElsbGVnYWwgUmVzb3VyY2UgQWNjZXNzfDN8IGZpbGVJZD0xODQwMDAwMTAwMDQzNzMxNjMgc291cmNlU2VydmljZU5hbWU9YXNzZXRzLnNiLmRhbnNrZXNwaWwuZGsgc2l0ZWlkPTc1MzQ0OTYgc3VpZD0yMDY2NjEyIHJlcXVlc3RDbGllbnRBcHBsaWNhdGlvbj1Nb3ppbGxhLzUuMCAoV2luZG93cyBOVCAxMC4wOyBXaW42NDsgeDY0KSBBcHBsZVdlYktpdC81MzcuMzYgKEtIVE1MLCBsaWtlIEdlY2tvKSBDaHJvbWUvNTkuMC4zMDcxLjExNSBTYWZhcmkvNTM3LjM2IGRldmljZUZhY2lsaXR5PWZyYSBjczI9ZmFsc2UgY3MyTGFiZWw9SmF2YXNjcmlwdCBTdXBwb3J0IGNzMz1mYWxzZSBjczNMYWJlbD1DTyBTdXBwb3J0IGNzMT1OQSBjczFMYWJlbD1DYXAgU3VwcG9ydCBjczQ9YTkzZDVjODktODZiMC00NzdiLTk4OGItYTJlMzhhYmU0N2Y4IGNzNExhYmVsPVZJRCBjczU9OWVjZmFlY2Q1NjM3ODk1YWExMDA2MDRjMWVlZGZmMGY4NDE3NzExYTNmZmI2OGQzZTI0NzdlZDI4MzMxOWNjYmQ0YWU0NzhlNmU5MTI1NjJlNmY1YTllMGNmZGRmMzFiZGE5OGM1NWJmY2Q2YWFlNzExYzBhYTI0YzMwYTFhOTQ4MmE2NGM2YTBkY2MxZTk0NjY4NjI1NDNmYzA4YzQyMDJlZmZmZTA0MmU4MTdjNTMzYmE2YzQ5YmYxNWI4M2M3MTExNDI1NDVjZmRlNzgwYjFjZjQ4MzM4YjJkM2Q3YjA3ODU2YWRhNThkY2E5NWI2YjllN2Q2NWI5OTFjYjE4MiBjczVMYWJlbD1jbGFwcHNpZyBkcHJvYz1Ccm93c2VyIGNzNj1DaHJvbWUgY3M2TGFiZWw9Y2xhcHAgY2NvZGU9QVQgY2ljb2RlPVZpZW5uYSBjczc9NDguMjA0OSBjczdMYWJlbD1sYXRpdHVkZSBjczg9MTYuMzY2MiBjczhMYWJlbD1sb25naXR1ZGUgQ3VzdG9tZXI9RGFuc2tlIFNwaWwgc3RhcnQ9MTcxMjA0MTY2MzUxNiByZXF1ZXN0PWFzc2V0cy5zYi5kYW5za2VzcGlsLmRrL2V0Yy9wcm9maWxlIHJlcXVlc3RNZXRob2Q9R0VUIGFwcD1IVFRQIGFjdD1SRVFfQkxPQ0tFRF9DT09LSUVMRVNTX1NFU1NJT04gZGV2aWNlRXh0ZXJuYWxJZD03MzExMTQyNDUxNDEzMDY1IGNwdD0zODg0NiBzcmM9ODEuMTgxLjU2LjEyMyBlbmQ9MTcxMjA0MTY2MzUxNyBmaWxlVHlwZT0zMDcwNDEgZmlsZVBlcm1pc3Npb249MCBjczk9IGNzOUxhYmVsPVJ1bGUgbmFtZSJ9LCJpc19qc29ubCI6ZmFsc2UsImVycm9yIjpudWxsfQ%3D%3D) using as log with a `cs7Label`.

**Comment by pront** [comment]
Also, yes you are correct those fields are not supported by the current CEF implementation. 

Can you point me to the specification you are using for reference? If those are part of the spec, it is worth creating an issue requesting enhancement. And as always, contributions are always welcome.

**Comment by aacgood** [comment]
Thanks @pront,

Im not sure of the actual spec that the vendor used in this case for their logging, however there are a large number of cs#Label/cs# fields with the data im working with.  My understanding is that CEF isnt limited to the number of these cs# fields that can be included.

Im not sure if this function can be updated to be more dynamic if the key starts with `cs#Label` and has as an associated `cs#` value that its able to be joined together like the existing fields.

Unfortunately I cant code in the language VRL is in so I wont be able to contribute a fix. 

[VRL Playground](https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLmNlZl93aXRoX2N1c3RvbV9maWVsZHMgPSBwYXJzZV9jZWYhKC5tZXNzYWdlLCB0cmFuc2xhdGVfY3VzdG9tX2ZpZWxkczogdHJ1ZSlcbi5jZWZfbm9fY3VzdG9tX2ZpZWxkcyA9IHBhcnNlX2NlZiEoLm1lc3NhZ2UpIiwiZXZlbnQiOnsibWVzc2FnZSI6IjwxMTQ%2BMjAyNC0wMS0wMVQwMDowMDowMC4wMDBaIE1ZSE9TVDAxIENFRjowfFZlbmRvcnxQcm9kdWN0fDIwMjQuNy4wLjJ8REVURUNUX0FORF9SRVNQT05EfEVORFBPSU5UfDl8Y2F0PVwiRW5kcG9pbnRcIiBjczJMYWJlbD1pbmNpZGVudElEIGNzMj0wMDAwMDAwMC0wMDAwLTAwMDAtMDAwMC0wMDAwMDAwMDAwMDBcIiBjczY3TGFiZWw9Y2FsbGJhY2tVUkwgY3M2Nz1cImh0dHBzOi8vbXkuZG9tYWluLmdvZXMuaGVyZS5jb20vIy9pbmNpZGVudHM%2FaW5jaWRlbnRJRFxcPTAwMDAwMDAwMC0wMDAwLTAwMDAtMDAwMC0wMDAwMDAwMDAwMDBcIiBjczY4TGFiZWw9dGVuYW50SUQgY3M2OD1JTlRFUk5BTCAifSwiaXNfanNvbmwiOmZhbHNlLCJlcnJvciI6bnVsbH0%3D)
